build/kokoro/build.sh

```diff
@@ -0,0 +1,163 @@
+#!/bin/bash
+# Copyright 2022 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -ex
+cd "$(dirname "$0")"
+
+
+##
+#
+# Environment Variables (required):
+#
+# JOB_MODE: 'integration' for stable or 'integration-dev' for adhoc experimental
+# LANGUAGE: 'java' | 'go' | 'cpp' | 'interop'
+#
+##
+
+PROJECT=microsvcs-testing
+CLUSTER=grpc-o11y-integration-testing-cluster
+ZONE=us-central1-c
+RELEASE=1.53.0-dev
+TEST_DIR=`realpath ../../observability/test`
+
+if [ -z "${JOB_MODE}" ] || [ -z "${LANGUAGE}" ] ; then
+  echo "Error: env var JOB_MODE and LANGUAGE are required."
+  exit 1
+fi
+
+if [ -z "${KOKORO_ARTIFACTS_DIR}" ] ; then
+  echo "Error: env var KOKORO_ARTIFACTS_DIR is not set."
+  exit 1
+fi
+
+
+
+##
+#
+# Install pre-requisites and set up environment
+#
+##
+
+gcloud config set project ${PROJECT}
+
+sudo apt-get -qq update
+sudo apt-get -qq install -y \
+  google-cloud-sdk-gke-gcloud-auth-plugin \
+  kubectl clang
+
+# Install Python packages
+python3 --version
+python3 -m pip install -r requirements.txt
+
+# Install Go
+curl -sSL https://go.dev/dl/go1.17.13.linux-amd64.tar.gz -o /tmp/go1.17.13.linux-amd64.tar.gz
+sudo tar -C /usr/local -xzf /tmp/go1.17.13.linux-amd64.tar.gz
+export PATH=${PATH}:/usr/local/go/bin
+
+# Configure GKE and docker auth
+export USE_GKE_GCLOUD_AUTH_PLUGIN=True
+gcloud container clusters get-credentials ${CLUSTER} --zone ${ZONE}
+gcloud auth configure-docker
+
+# TODO(stanleycheung): find a more appropriate dir
+export REPOS_BASE_DIR=${KOKORO_ARTIFACTS_DIR}/github
+
+
+
+##
+#
+# Build TestService client/server binaries for each language
+#
+##
+
+build_java () {
+  mkdir -p ${REPOS_BASE_DIR}/grpc-java
```

Comment:
In general, I would recommend doing builds within docker images for hermeticity. Kokoro images can change underneath you.

Reply:
This has been implemented. Most of the build instructions have been done in individual language's `Dockerfile`. There is still a call out to the entry point script in individual language repo here. I plan to improve that in the future.